pyvista/plotting/tools.py

```diff
@@ -459,33 +460,95 @@ def opacity_transfer_function(mapping, n_colors, interpolate=True,
+
+    If ``color`` is a sequence of RGBA floats, the ``opacity`` parameter
+    is ignored.
 
     Values returned will be between 0 and 1.
 
+    Parameters
+    ----------
+    color : str or sequence
+        Either a string, RGB sequence, RGBA sequence, or hex color string.
+        RGB(A) sequences should only contain values between 0 and 1.
+        For example:
+
+            * ``'white'``
+            * ``'w'``
+            * ``[1, 1, 1]``
+            * ``[0.5, 1, 0.7, 1]``
+            * ``'#FFFFFF'``
+
+    opacity : float, optional
+        Default opacity of the returned color. Used when ``color`` is
+        not a length 4 RGBA sequence. Only opacities between 0 and 1
+        are allowed.
+
+    default_color : str or sequence, optional
+        Default color to use when ``color`` is None.  If this value is
+        ``None``, then defaults to the global theme color.  Format is
+        identical to ``color``.
+
+    Returns
+    -------
+    tuple
+        Either a length 3 RGB sequence if opacity is unset, or an RGBA
+        sequence when ``opacity`` is set or the input ``color`` is an
+        RGBA sequence.
+
+    Examples
+    --------
+    >>> import pyvista
+    >>> pyvista.parse_color('blue')
+    (0.0, 0.0, 1.0)
+
+    >>> pyvista.parse_color('k')
+    (0.0, 0.0, 0.0)
+
+    >>> pyvista.parse_color('#FFFFFF')
+    (1.0, 1.0, 1.0)
+
+    >>> pyvista.parse_color((0.4, 0.3, 0.4, 1))
+    (0.4, 0.3, 0.4, 1.0)
+
     """
+    color_valid = True
     if color is None:
         if default_color is None:
             color = pyvista.global_theme.color
         else:
             color = default_color
     if isinstance(color, str):
         color = string_to_rgb(color)
-    elif len(color) == 3:
-        pass
-    elif len(color) == 4:
-        color = color[:3]
+    elif isinstance(color, (Sequence, np.ndarray)):
+        try:
+            color = np.asarray(color, dtype=np.float64)
+            if color.ndim != 1 or color.size not in (3, 4) or not np.all((0 <= color) & (color <= 1)):
+                color_valid = False
+            elif len(color) == 4:
+                opacity = color[3]
+                color = color[:3]
+        except ValueError:
+            color_valid = False
     else:
+        color_valid = False
+    if not color_valid:
         raise ValueError(f"""
     Invalid color input: ({color})
     Must be string, rgb list, or hex color string.  For example:
         color='white'
         color='w'
         color=[1, 1, 1]
         color='#FFFFFF'""")
-    if opacity is not None and isinstance(opacity, (float, int)):
-        color = [color[0], color[1], color[2], opacity]
-    return color
+    if opacity is not None:
+        if isinstance(opacity, (float, int)) and 0 <= opacity <= 1:
+            color = [color[0], color[1], color[2], float(opacity)]
+        else:
+            raise ValueError(f"""
+    Invalid opacity input: {opacity}
+    Must be a scalar value between 0 and 1.""")
```

Comment:
Now this is really just nitpicking, but we usually handle multiline error messages with implicit line continuations:
https://github.com/pyvista/pyvista/blob/0c344920d6959205aaa251bdb190b05c1487a1c1/pyvista/core/filters/data_set.py#L4341-L4344
or sometimes with `textwrap.dedent`:
https://github.com/pyvista/pyvista/blob/0c344920d6959205aaa251bdb190b05c1487a1c1/pyvista/plotting/plotting.py#L4778-L4783

I personally dislike letting strings "underflow" into lower indentation levels.